Elemental/Components/AeTable/AeBaseTable.cs

```diff
@@ -8,119 +8,127 @@
-      }
-      else
-      {
-        return Dataset.OrderByDescending(data => Accessors[_sorting.column].Invoke(data));
-      }
-    }
+        protected IEnumerable<T> Rows { get => GetSortedDataset(); }
+        protected (int column, bool sortAscending) _sorting = (0, true);
 
-    protected virtual string CalculateGridTemplateColumns()
-    {
-      if(string.IsNullOrWhiteSpace(GridTemplateColumns)){
-        return $"repeat({(Accessors?.Count ?? 1)}, 1fr)";
-      }
 
-      return GridTemplateColumns;
-    }
+        protected string _tableClass => $"ae table {_inputClass}";
+        protected string _gridStyle => $"grid-template-columns: {CalculateGridTemplateColumns()};";
 
-    protected virtual void ToggleSortOnColumn(int column)
-    {
-        if (column == _sorting.column)
+        protected override void OnInitialized()
         {
-            _sorting.sortAscending = !_sorting.sortAscending;
+            base.OnInitialized();
         }
-        else
+
+        protected virtual IEnumerable<T> GetSortedDataset()
         {
-            _sorting.column = column;
-            _sorting.sortAscending = true;
+            if (Dataset is null || !Dataset.Any())
+            {
+                return new List<T>();
+            }
+
+            if (_sorting.sortAscending)
+            {
+                // if Sorter Parameters exist, use sorter functions, else use Accessors TableComparer
+                return (Sorters != null && Sorters[_sorting.column] != null)?
+                    Dataset.OrderBy(data => data, Sorters[_sorting.column]) :
+                    Dataset.OrderBy(data => Accessors[_sorting.column].Invoke(data));
+            }
+            else
+            {
+                return (Sorters != null && Sorters[_sorting.column] != null) ?
+                    Dataset.OrderByDescending(data => data, Sorters[_sorting.column]) :
+                    Dataset.OrderByDescending(data => Accessors[_sorting.column].Invoke(data));
+            }
         }
-        StateHasChanged();
-    }
 
-    protected virtual string GetSortIconClass(int column)
-    {
-        if (column == _sorting.column)
+        protected virtual string CalculateGridTemplateColumns()
         {
-            return _sorting.sortAscending ? "fad fa-sort-down" : "fad fa-sort-up";
+            if (string.IsNullOrWhiteSpace(GridTemplateColumns))
+            {
+                return $"repeat({(Accessors?.Count ?? 1)}, 1fr)";
+            }
+
+            return GridTemplateColumns;
         }
-        return "fad fa-sort";
-    }
 
-    // It's not recommended to use the RenderFragment builder yet so this is a catch-all workaround
-    protected virtual RenderFragment DefaultHeader(string header, int index){
-      throw new NotImplementedException("You must implement the DefaultHeader method for your Table Component.");
-    }
+        protected virtual void ToggleSortOnColumn(int column)
+        {
+            Console.WriteLine("ToggleSortOnColumn");
```

Comment:
👀

Reply:
👀👀

Reply:
I somehow completely missed and forgot about that! 😅